protocol/clproto_cpp/src/encoders.cpp

```diff
@@ -218,4 +222,51 @@ proto::AnalogIOState encoder(const AnalogIOState& io_state) {
   *message.mutable_values() = matrix_encoder(io_state.data());
   return message;
 }
+
+proto::CartesianTrajectory encoder(const CartesianTrajectory& trajectory) {
+  proto::CartesianTrajectory message;
+  *message.mutable_state() = encoder(static_cast<State>(trajectory));
+  *message.mutable_reference_frame() = trajectory.get_reference_frame();
+  if (trajectory.is_empty()) {
+    return message;
+  }
+  std::vector<std::string> names;
+  std::vector<uint64_t> durations;
+  for (unsigned int i = 0; i < trajectory.get_size(); ++i) {
+    auto [point, duration] = trajectory[i];
+    auto vec = point.data();
+    auto data = message.add_data();
+    data->mutable_values()->Assign(vec.data(), vec.data() + vec.size());
+    names.push_back(point.get_name());
+    durations.push_back(duration.count());
+  }
+  message.mutable_names()->Assign(names.begin(), names.end());
+  message.mutable_durations()->Assign(durations.begin(), durations.end());
```

Comment:
nitpick: you're using `Assign` in these places but the pattern everywhere else has been to use the `=` assignment operator instead, e.g.
```
message.mutable_names() = {names.begin(), names.end()};
```

Is there a reason for preferring Assign? If it's functionally equivalent I would prefer to stick with the consistent prior pattern, but if it's somehow more efficient then I would rather propose a followup refactor applied in all places.

Reply:
no particular reason, no. As far as I understand `{}` should just be a shorthand. I'll make the change